gptme/tools/base.py

```diff
@@ -240,3 +302,37 @@ def _to_xml(self) -> str:
         args = " ".join(self.args)
         args_str = "" if not args else f" args='{args}'"
         return f"<tool-use>\n<{self.tool}{args_str}>\n{self.content}\n</{self.tool}>\n</tool-use>"
+
+
+@dataclass(frozen=True)
+class APIToolUse(BaseToolUse):
+    tool: str
+    args: dict[str, str]
```

Comment:
I think we should be able to solve this without subclassing. We could support both `args` (list) and `kwargs` (dict) in the same ToolUse class.

Reply:
Done.